### DISCUSSION TITLE
bpf2go in vendor mode

### DISCUSSION THREAD

**Discussion by lec-bit** [cilium/ebpf#1090] [Q&A] [answered]
When I use bpf2go of the latest version v0.11.0 to parse the ebpf file, I encounter the following problem:
The bpf2go file in cilium/ebpf/ on GOPATH can be parsed normally. The bpf2go file can also be parsed normally by using go build to compile binary file bpf2go. However, if I run go mod vendor to generate the vendor directory, The cmd/bpf2go directory does not exist in the vendor directory. After the cmd/bpf2go directory is manually copied from the GOPATH environment, the cmd/bpf2go directory cannot be parsed and an error is reported.

Error: can't write kmesh /home/gitee/fork/last-version/Kmesh/bpf/kmesh/bpf2go/kmeshcgroupsock_bpfel.go: 12:2: invalid import path: "": ""
exit status 1

The check result shows that the file is in ebpf/internal/output.go.
Formatted, err := Error when format.Source(src)

Strangely enough, you can manually copy it from an earlier version of cilium/ebpf.

I want to know why the vendor directory generated by go mod vendor is missing files, and why the cilium/ebpf of v0.11.0 cannot be copied directly compared to the old version.

**Comment by lmb** [comment, marked as answer]
This might be due to https://github.com/cilium/ebpf/commit/223b7d740c167a41a7ba82ad59f81c7ef375b843. Maybe `debug.BuildInfo` returns something different when in vendor mode? Can you try extending the fallback in https://github.com/cilium/ebpf/commit/b04b7582fa60f0af87aac744a63cabe2ffe082e6 to cover the case where the import path is `""`?

**Reply by lec-bit** [reply]
Thank you very much for your help
You're right, debug.ReadBuildInfo returns true but no BuildInfo when in vendor, and that led to this error.
I don't know if debug.ReadBuildInfo returns like this is his mechanism, I'm using it wrong or if it's buggy. If so, does the current ciluim/ebpf/cmd/bpf2go need to be adjusted to support vendors?

**Reply by lmb** [reply]
Yes, I think two things should happen:
1. Could you file a bug against upstream Go `runtime/debug` which outlines the problem?
2. Can you send a patch which extends the fall-back behaviour when bi.Main.Path is empty?
